include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -165,6 +165,44 @@ class CommonOperatorLoader : public ProtobufLoader {
     }
   }
 
+  void loadReshape(const OpType &op, ArgumentDictionaryTy &dict) {
+    const std::string &opName = loadOperatorName(op);
+    auto *in = getOrCreateNodeByName(op.input(0));
+
+    std::vector<size_t> newDim;
+    if (dict.count("shape")) {
+      std::vector<int64_t> protoDims = getShape<int64_t>(dict["shape"]);
+
+      auto oldDim = in->dims();
+      int64_t product = 1;
+      for (size_t i = 0, e = protoDims.size(); i != e; i++) {
+        if (protoDims[i] == 0)
+          // shape[i] == 0 means that corresponding element should remain
+          // the same.
+          protoDims[i] = oldDim[i];
+        if (protoDims[i] != -1)
+          product *= protoDims[i];
+      }
+      for (size_t i = 0, e = newDim.size(); i != e; i++) {
+        if (protoDims[i] == -1)
+          // shape[i] == -1 means that corresponding element should be inferred
```

Comment:
Does Caffe2 do the same thing as ONNX here?

If not, might be a good idea to assert which backend expects this, otherwise we may handle one incorrectly and create an annoying bug to catch.

Reply:
Yes, it's the same for both Caffe2 and ONNX.